woocommerce/wc-admin
# Fix bar chart in narrow viewports

Fixes #1190.

The bar chart was empty when there were many bars to be displayed and the viewport was too narrow.

By default, each date took at least 3 pixels: 1 for each bar and 1 as a padding between dates. When displaying an entire year, that meant that if the available space was less than 1095px (365*3), each bar would have less than a pixel, so they were not visible because they were 0 pixels wide.

This PR fixes that with two approaches:
* Eliminates the padding between dates when the available space is too narrow. When displaying an entire year, that will be 730px-1095px.
* When the available space is so narrow that removing the padding is not enough, allows the chart to take the minimum needed space and makes it scrollable.

### Screenshots
![image](https://user-images.githubusercontent.com/3616980/50599486-8a5d9d00-0eae-11e9-819e-2f11a89a1d79.png)

### Detailed test instructions:
- Go to the _Revenue_ report or any other.
- Select a wide date range (for example, _Last year_).
- Select bars as the chart type.
- Make the viewport wider and narrower and verify bars are always visible.
- Make sure the tooltip is shown in the correct position when hovering the bars.
